db/blob/blob_source.h

```diff
@@ -52,25 +53,45 @@ class BlobSource {
                  FilePrefetchBuffer* prefetch_buffer, PinnableSlice* value,
                  uint64_t* bytes_read);
 
-  // Read multiple blobs from the underlying cache or storage.
+  // Read multiple blobs from the underlying cache or on-disk blob file(s).
   //
-  // If successful, returns ok and sets the elements of blobs to the newly
+  // If successful, returns ok and sets the elements of "blob_reqs" to the newly
   // retrieved uncompressed blobs. If there was an error while fetching one of
-  // blobs, sets its corresponding "blobs[i]" to empty and sets "statuses[i]" to
-  // a non-ok status.
+  // blobs, sets its corresponding "blob_reqs[i].result" to empty and sets
+  // "blob_reqs[i].status" to a non-ok status.
   //
   // Note:
-  //  - Offsets must be sorted in ascending order by caller.
+  //  - The main difference between this function and MultiGetBlobFromOneFile is
+  //    that this function can read multiple blobs from multiple blob files.
+  //
+  //  - For consistency, whether the blob is found in the cache or on disk, sets
+  //  "*bytes_read" to the total size of on-disk (possibly compressed) blob
+  //  records.
+  void MultiGetBlob(const ReadOptions& read_options,
+                    std::map</*blob file number*/ uint64_t,
```

Comment:
I need to figure out why `unordered_map` fails the `DBBlobBasicTest/DBBlobBasicIOErrorMultiGetTest.MultipleBlobFiles` test. It appears that the files must be retrieved in ascending order, or the code may have other flaws.

Reply:
Thanks for clarifying!!

Reply:
After we replace `std::map` with `std::tuple`, we don't have this problem now.